src/impl/tpls/KokkosBlas1_iamax_tpl_spec_decl.hpp

```diff
@@ -699,4 +391,193 @@ KOKKOSBLAS1_CIAMAX_TPL_SPEC_DECL_CUBLAS(unsigned int, Kokkos::LayoutLeft,
+        const int XST = X.stride(0);                                           \
+        const int LDX = (XST == 0) ? 1 : XST;                                  \
+        int idx;                                                               \
+        KokkosBlas::Impl::RocBlasSingleton& s =                                \
+            KokkosBlas::Impl::RocBlasSingleton::singleton();                   \
+        rocblas_pointer_mode prevPtrMode;                                      \
+        KOKKOS_ROCBLAS_SAFE_CALL_IMPL(                                         \
+            rocblas_get_pointer_mode(s.handle, &prevPtrMode));                 \
+        if (prevPtrMode == rocblas_pointer_mode_device) {                      \
+          KOKKOS_ROCBLAS_SAFE_CALL_IMPL(                                       \
+              rocblas_set_pointer_mode(s.handle, rocblas_pointer_mode_host));  \
+        }                                                                      \
+        KOKKOS_ROCBLAS_SAFE_CALL_IMPL(                                         \
+            ROCBLAS_FN(s.handle, N,                                            \
+                       reinterpret_cast<const ROCBLAS_SCALAR_TYPE*>(X.data()), \
+                       LDX, &idx));                                            \
+        R() = static_cast<size_type>(idx);                                     \
+        if (prevPtrMode == rocblas_pointer_mode_device) {                      \
+          KOKKOS_ROCBLAS_SAFE_CALL_IMPL(rocblas_set_pointer_mode(              \
+              s.handle, rocblas_pointer_mode_device));                         \
+        }                                                                      \
+      } else {                                                                 \
+        Iamax<RV, XV, 1, false, ETI_SPEC_AVAIL>::iamax(R, X);                  \
+      }                                                                        \
+      Kokkos::Profiling::popRegion();                                          \
+    }                                                                          \
+  };                                                                           \
+  template <class ExecSpace>                                                   \
+  struct Iamax<                                                                \
+      Kokkos::View<INDEX_TYPE, LAYOUT, Kokkos::Device<ExecSpace, MEMSPACE>,    \
+                   Kokkos::MemoryTraits<Kokkos::Unmanaged> >,                  \
+      Kokkos::View<const SCALAR_TYPE*, LAYOUT,                                 \
+                   Kokkos::Device<ExecSpace, MEMSPACE>,                        \
+                   Kokkos::MemoryTraits<Kokkos::Unmanaged> >,                  \
+      1, true, ETI_SPEC_AVAIL> {                                               \
+    typedef Kokkos::View<INDEX_TYPE, LAYOUT,                                   \
```

Comment:
Please add a top-level macro or update this macro to handle the two host & device overloads:
```
KOKKOSBLAS1_XIAMAX_TPL_SPEC_DECL_ROCBLAS(... Kokkos::Device<ExecSpace, MEMSPACE>)
KOKKOSBLAS1_XIAMAX_TPL_SPEC_DECL_ROCBLAS(... Kokkos::HostSpace)
```

Reply:
@e10harvey I added top-level macros for rocBLAS and cuBLAS iamax.